src/libs/ajax/leonardo/providers/LeoRuntimeProvider.ts

```diff
@@ -0,0 +1,28 @@
+import { isGcpContext } from 'src/analysis/utils/runtime-utils';
+import { Ajax } from 'src/libs/ajax';
+import { ListRuntimeItem, Runtime } from 'src/libs/ajax/leonardo/models/runtime-models';
+import { RuntimeWrapper } from 'src/libs/ajax/leonardo/Runtimes';
+
+export type RuntimeBasics = Pick<Runtime, 'runtimeName' | 'googleProject' | 'cloudContext'>;
+
+export interface LeoRuntimeProvider {
+  list: (listArgs: Record<string, string>, signal?: AbortSignal) => Promise<ListRuntimeItem[]>;
+  stop: (runtime: RuntimeWrapper) => Promise<void>;
+  delete: (runtime: Runtime, workspaceId: string, deleteDisk: boolean) => Promise<void>;
+}
+
+export const leoRuntimeProvider: LeoRuntimeProvider = {
+  list: (listArgs: Record<string, string>, signal?: AbortSignal): Promise<ListRuntimeItem[]> => {
+    return Ajax(signal).Runtimes.listV2(listArgs);
+  },
+  stop: (runtime: RuntimeWrapper, signal?: AbortSignal): Promise<void> => {
+    return Ajax(signal).Runtimes.runtimeWrapper(runtime).stop();
```

Comment:
perhaps it is worth removing `runtimeWrapper` from `Runtimes`, since the type `RuntimeWrapper` is more or less the same as `RuntimeBasics` and `Runtimes` has the logic the provider has within the `runtimeWrapper`

Reply:
I held off on a more dramatic change since quite a few consumers would be touched by changes to Ajax.Runtimes.runtimeWrapper pattern....  I am thinking this could be it's own PR to re-align all consumers to LeoRuntimeProvider and shift what's currently happening in Ajax layer to happen in this provider instead.

Reply:
Yeah, I think a TODO to align `RuntimeWrapper` and `RuntimeBasics` would be sufficient

Reply:
added TODO